pages/components/Button.tsx

```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import styles from "@/styles/Home.module.scss";
+
+type ButtonProps = {
+  text: string;
+  className: string;
+  onClick?: () => void;
+};
```

Comment:
```suggestion
interface ButtonProps extends HTMLAttributes<HTMLButtonElement>  {
  text: string;
  className: string;
  onClick?: () => void;
};
```
버튼이나 인풋같은 기본적인 컴포넌트들은 해당 돔에서 사용하는 이벤트를 그대로 상속해주는게 좋아요. 이는 객체지향에서 LSP(리스코프 치환 원칙)이라고 하는데 간단하게 설명드리면 제가 button이라는 html을 사용할때 쓰는 인터페이스를 그대로 Button컴포넌트에서 사용하면 유지보수나 사용성에서 좋겠죠?
그래서 버튼같은 근본 컴포넌트는 위처럼 상속해서 쓰는편이에요. 만약 해당 버튼에 onTouch같은 함수들이 필요할때 매번 넣어주면 불편하겠죠 >?